### ISSUE TITLE
bpls -d throws

### ISSUE THREAD

**Issue by olesenm** [ornladios/ADIOS2#683]
I've restarted my effort from last Nov to begin migration from ADIOS1 to ADIOS2 (lost some time with API changes and other things along the way).
I'm still trying to figure out why I'm not really generating data properly (files are too small) and thus figured on using `bpls` to try to understand what is where. `bpls -l file` shows me my variables and attributes, but using the `-d`  or `--dump` core dumps with this type of throw:
```
terminate called after throwing an instance of 'std::invalid_argument'
  what():  ERROR: selection is not valid for single value variable v1, in call to SetSelection
```

I've reached all the way back and now started using the `examples/basic/localArray` etc, but in all cases I get the same core dump with `bpls`.

Is there any other way to diagnose my file contents?

/mark

**Comment by williamfgc** [comment]
Hi @olesenm please use adios1's bpls as a quick work around. I am still working on bugs and sorting out things for this month's (June 30th) release.  BTW, local arrays are still not fully supported on the read side. I'd stick with global arrays for now. Also, we are figuring out ways to provide the memory from the engine itself. We'll keep you posted.

**Comment by olesenm** [comment]
OK, I'll give it a try with bp1 tomorrow. Not sure how global arrays really fit into our usual storage, but will cross that bridge later.

**Comment by williamfgc** [comment]
@olesenm that's a good question. On the reader side if either adios2 or the user provides pre-allocated memory it will be the same footprint. We do this since the low-level API `Get` function requires pre-allocated memory, while the high-level API (fstream based, not for performance, but analysis) `read` returns a vector taking advantage of move semantics. The challenge is for a write buffer, basically doing an allocation and returning the memory to the user to be filled. The latter is what you requested, it won't be in the June release, but we'll be working on it right after. I'll let you know when it's in master.

**Comment by williamfgc** [comment]
about global vs local, just make the shape = count and start = zeros to get a global variable. e.g. `DefineVariable<T>("var", {5,5,5}, {0,0,0}, {5,5,5}, ...);  Truly local variables (local to a rank) are a special case in MPI runs and they are retrieve as "blocks". In fact, for non-MPI there is no distinction. We are prioritizing global variables functionality since this is our common use case.

**Comment by pnorbert** [comment]
This is being fixed in the upcoming release 2.3.1
